File: Python/Data Structures and Algorithms/Linear Shoal/ArrayI/ShuffleTheArray.py
""" Given the array nums consisting of 2n elements in the form [x1,x2,...,xn,y1,y2,...,yn].

Return the array in the form [x1,y1,x2,y2,...,xn,yn]. """

class Solution:
    def shuffle(self, nums: list[int], n: int) -> list[int]:
        ans = [0] * (2 * n)

        for x in range(0, n):
            ans[x * 2] = nums[x]
            ans[(x * 2) + 1] = nums[n + x]
            
        return ans

File: Cpp/Data Structures and Algorithms/Linear Shoal/Array I/ShuffleTheArray.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> ans(2 * n);

        for (int i = 0; i < n; i++)
        {
            ans[i*2] = nums[i];
            ans[i*2 + 1] = nums[i + n];
        }
        
        return ans;
    }
};

/* 100%:

class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> temp;
        for(int i = 0; i < n; ++i){
           temp.push_back(nums[i]);
           temp.push_back(nums[i+n]); 
        }
        return temp;
    }
}; */